lightning-persister/src/util.rs

```diff
@@ -39,7 +39,10 @@ fn path_to_windows_str<T: AsRef<OsStr>>(path: T) -> Vec<winapi::shared::ntdef::W
 }
 
 #[allow(bare_trait_objects)]
-pub(crate) fn write_to_file<D: DiskWriteable>(path: PathBuf, filename: String, data: &D) -> std::io::Result<()> {
+pub(crate) fn write_to_file<D: DiskWriteable>(full_filepath: String, data: &D) -> std::io::Result<()> {
+	let full_filepath = PathBuf::from(full_filepath);
+	let path = full_filepath.parent().unwrap();
+	let filename = full_filepath.file_name().unwrap().to_str().unwrap().to_string();
```

Comment:
bleh. Can we not at least leave this an `str` or maybe split on a `/` manually instead of going through `PathBuf`? Are there any platforms that use '\' for paths anymore?

Reply:
removed need for PathBuf here -- the original implementation took in a PathBuf which is why I tried to 'put it back'